presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZAddressRange.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.zorder;
+
+import java.util.Objects;
+
+/**
+ * The ZAddressRange class contains a minimum and maximum address. Both are inclusive.
+ */
+public class ZAddressRange
+        implements Comparable<ZAddressRange>
+{
+    private final long minimumAddress;
+    private final long maximumAddress;
+
+    public ZAddressRange(long minimumAddress, long maximumAddress)
+    {
+        this.minimumAddress = minimumAddress;
+        this.maximumAddress = maximumAddress;
+    }
+
+    public long getMinAddress()
+    {
+        return minimumAddress;
+    }
+
+    public long getMaxAddress()
+    {
+        return maximumAddress;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof ZAddressRange)) {
+            return false;
+        }
+        ZAddressRange that = (ZAddressRange) o;
+        return minimumAddress == that.minimumAddress && maximumAddress == that.maximumAddress;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(minimumAddress, maximumAddress);
+    }
+
+    @Override
+    public int compareTo(ZAddressRange that)
```

Comment:
I feel it is not quite intuitive to define the priority of range comparison to be "min first, if equal then max". If this is used somewhere, maybe we should consider using lambda to create an anonymous `Comparator`?

Reply:
The function is not needed in this current implementation, but when supporting multiple query ranges (e.g. col <= 0 or col >= 5), I find the addressRanges for each input range, then sort the list in that order.
I only sort once, though, so I can use a lambda function instead 👍